src/systemctl/systemctl-list-unit-files.c

```diff
@@ -103,7 +114,9 @@ static int output_unit_file_list(const UnitFileList *units, unsigned c) {
                 else
                         on_unit_color = on_underline;
 
-                id = basename(u->path);
+                r = path_extract_filename(u->path, &id);
```

Comment:
actually there's a nicer solution here. just add u->path to the table below as `TABLE_PATH_BASENAME` instead of `TABLE_STRING`, and the right thing will happen.